src/main/java/uk/gov/companieshouse/authcodenotification/controller/AuthCodeNotificationController.java

```diff
@@ -51,18 +57,35 @@ public ResponseEntity<Object> sendEmail(@RequestHeader(value = ERIC_REQUEST_ID_K
         var authCode = dataSanitiser.makeStringSafe(sendEmailRequestDto.getAuthCode());
 
         var logDataMap = new DataMap.Builder().companyNumber(companyNumber).build();
-        ApiLogger.infoContext(requestId,"Request received for auth code email", logDataMap.getLogMap());
+        var logMap = logDataMap.getLogMap();
+        ApiLogger.infoContext(requestId, "Request received for auth code email", logMap);
 
+        // validate request data
         var validationErrors =  authCodeEmailValidator.validate(companyNumber, authCode, new Errors(), requestId);
         if (validationErrors.hasErrors()) {
             ApiLogger.errorContext(requestId, String.format(VALIDATION_ERRORS_MESSAGE,
-                    convertErrorsToJsonString(validationErrors)), null, logDataMap.getLogMap());
+                    convertErrorsToJsonString(validationErrors)), null, logMap);
             var responseBody = ChResponseBody.createErrorsBody(validationErrors);
             return new ResponseEntity<>(responseBody, HttpStatus.BAD_REQUEST);
         }
 
+        // encrypt auth code
+        String authCodeEncrypted;
+        try {
+            authCodeEncrypted = encrypter.encrypt(authCode);
```

Comment:
Is the (web) controller code the right place to do this encryption? Sounds more like something that should be buried in the service implementation.

Reply:
good point, do you think it would be better if it happened inside the AuthCodeNotificationService.sendAuthCodeEmail method or should it be a separate method in the service that is called by the controller before the controller calls sendAuthCodeEmail